update_python_packages.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import subprocess
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from zipfile import ZipFile
+
+
+def zip_recursively(zip_file, root_dir, folder_name):
+    for root, _, files in os.walk(root_dir / folder_name):
+        for file in files:
+            full_file_path = Path(os.path.join(root, file))
+            zip_file.write(full_file_path, full_file_path.relative_to(root_dir))
+
+
+with TemporaryDirectory() as temp_dir:
+    temp_dir = Path(temp_dir)
+
+    # Pip install everything and capture everything that was installed.
+    subprocess.run(
+        [
+            "python2",
+            "-m",
+            "pip",
+            "install",
+            "-r",
+            "requirements.txt",
+            "--no-compile",
+            # The combination of --target and --upgrade forces pip to install
+            # all packages to the temporary directory, even if an already existing
+            # version is installed
+            "--target",
+            temp_dir,
+            "--upgrade",
+        ]
+    )
+    subprocess.run(
+        ["python2", "-m", "pip", "freeze", "--path", temp_dir],
+        stdout=open("frozen_requirements.txt", "w"),
+    )
+
+    # Quickly compute the number of requirements we have.
+    nb_dependencies = len([_ for _ in open("frozen_requirements.txt", "rt")])
```

Comment:
Would this cause issues if we have comments in the frozen_requirements file? Though there shouldn't really, since it is auto generate.

Reply:
nm the assert further down would cause someone to look at it, and hopefully see the issue if there was.